File: frontend/src/types/note.ts
export interface Note {
  id: number;
  content: string;
  lead_id: number;
  organization_id: number;
  created_by_id: number;
  deal_id?: number;
  created_at: string;
  updated_at?: string;
}

export interface NoteCreate {
  content: string;
  lead_id: number;
  organization_id: number;
  deal_id?: number;
}

export interface NoteUpdate {
  content: string;
}

export interface NoteList {
  items: Note[];
  total: number;
}

File: backend/app/schemas/note.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime

# Note Base Schema
class NoteBase(BaseModel):
    content: str
    lead_id: int
    organization_id: int
    deal_id: Optional[int] = None

# Properties to receive via API on creation
class NoteCreate(NoteBase):
    created_by_id: int

# Properties to receive via API on update
class NoteUpdate(BaseModel):
    content: Optional[str] = None

# Properties stored in DB
class NoteInDB(NoteBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    created_by_id: int

    class Config:
        from_attributes = True

# Properties to return via API
class Note(NoteInDB):
    pass

# Properties for Note list
class NoteList(BaseModel):
    items: List[Note]
    total: int

    class Config:
        from_attributes = True
